src/file_lines.rs

```diff
@@ -144,6 +144,34 @@ impl FileLines {
             Err(_) => false,
         }
     }
+
+    pub fn contains_line(&self, file_name: &str, line: usize) -> bool {
+        let map = match self.0 {
+            // `None` means "all lines in all files".
+            None => return true,
+            Some(ref map) => map,
+        };
+
+        match canonicalize_path_string(file_name)
+                  .and_then(|canonical| map.get_vec(&canonical).ok_or(())) {
+            Ok(ranges) => ranges.iter().any(|r| r.lo <= line && r.hi >= line),
+            Err(_) => false,
+        }
+    }
+
+    pub fn intersects_range(&self, file_name: &str, lo: usize, hi: usize) -> bool {
```

Comment:
I think there is a lot that could be refactored out from these two functions - you could have a function `matches_range` or something (I hate naming :-) ) which takes a function for matching against a range and you pass it `|r| r.lo <= line && r.hi >= line` or `|r| r.intersects(Range::new(lo, hi))`.